mkdir_linux.go

```diff
@@ -22,6 +22,8 @@ var (
 	errPossibleAttack = errors.New("possible attack detected")
 )
 
+type FileMode = os.FileMode
```

Comment:
The problem with doing this (and this is the reason why I didn't use `os.FileMode` in the first place) is that the `os.FileMode` bits don't match the Unix bits outside of `0o777`. Passing the wrong mode bits is a common bug and accepting `os.FileMode` would lead to confusion unless we also added conversion code as well.

I think it would be simpler to just use `uint32` instead (though looking it up, it seems the kernel actually defines `umode_t` as `unsigned short` which is `uint16`).